src/agent/v8/inspector-debugapi.ts

```diff
@@ -132,46 +134,169 @@ export class InspectorDebugApi implements debugapi.DebugApi {
         utils.messages.INVALID_BREAKPOINT
       );
     }
-    const baseScriptPath = path.normalize(breakpoint.location.path);
-    if (!this.sourcemapper.hasMappingInfo(baseScriptPath)) {
-      const extension = path.extname(baseScriptPath);
-      if (!this.config.javascriptFileExtensions.includes(extension)) {
+
+    // breakpoint.location line & column are 1-based per API documentation.
+    let mapInfo = {
+      file: path.normalize(breakpoint.location.path),
+      line: breakpoint.location.line,
+      column: breakpoint.location.column || 1,
+    };
+
+    if (this.sourcemapper.hasMappingInfo(mapInfo.file)) {
+      // There is a sourcemap on the filesystem that refers to the requested file.
+      // mappingInfo expects 0-based line & column.
+      const mappedLocation = this.sourcemapper.mappingInfo(
+        mapInfo.file,
+        mapInfo.line - 1,
+        mapInfo.column - 1
+      );
+      if (mappedLocation) {
+        // mappingInfo returns 0-based line & column.  Map it back to 1-based.
+        mapInfo = {
+          file: mappedLocation.file,
+          line: mappedLocation.line + 1,
+          column: mappedLocation.column + 1,
+        };
+      }
+    }
+
+    // Try compiling the condition (if applicable).
+    const compile = utils.getBreakpointCompiler(breakpoint);
+    if (breakpoint.condition && compile) {
+      try {
+        breakpoint.condition = compile(breakpoint.condition);
+      } catch (e) {
+        this.logger.info(
+          'Unable to compile condition >> ' + breakpoint.condition + ' <<'
+        );
         return utils.setErrorStatusAndCallback(
           cb,
           breakpoint,
-          StatusMessage.BREAKPOINT_SOURCE_LOCATION,
-          utils.messages.COULD_NOT_FIND_OUTPUT_FILE
+          StatusMessage.BREAKPOINT_CONDITION,
+          utils.messages.ERROR_COMPILING_CONDITION
         );
       }
+    }
 
-      this.setInternal(breakpoint, null /* mapInfo */, null /* compile */, cb);
-    } else {
-      const line = breakpoint.location.line;
-      const column = 0;
-      const mapInfo = this.sourcemapper.mappingInfo(
-        baseScriptPath,
-        line,
-        column
+    // Presently it is not possible to precisely disambiguate the script
```

Comment:
The [config](https://github.com/googleapis/cloud-debug-nodejs/blob/master/src/agent/config.ts) has a `workingDirectory` configuration that could be useful.